src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -77,7 +87,21 @@ public partial class Activity
         /// <para/>
         /// See <see href="https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/ActivityUserGuide.md#id-format"/> for more details
         /// </summary>
-        public string ParentId { get; private set; }
+        public string ParentId
+        {
+            get
+            {
+                // if we represented it as a traceId-spanId, convert it to a string.  
+                if (_parentId == null)
+                {
+                    if (_parentSpanIdSet)
+                        _parentId = "00-" + _traceId.AsHexString + "-" + _parentSpanId.AsHexString + "-00";
```

Comment:
If we're going to have AsHexString create a StringBuilder anyway, it'd be nice to avoid doing that and creating each consituent string here, then the string[] to store all of these pieces for the concat, and then the final string... instead it'd be nicer to just create the StringBuilder (or ValueStringBuilder) here, pass it into AsHexString, and end up with just the resulting string allocation.  To use ValueStringBuilder, you can just include the relevant .cs file... that's what we do in other corefx projects right now, since it's not public.

Reply:
I do see that extra allocations here.   We can actually use the same stackalloc trick here (we actually to know it is fixed length),  However I would like to hold off because the theory is that under hot code paths we don't use the 'Id' property (we ask for SpanId and TraceIds).   Thus if we do our job right, this path will not be 'hot' and thus worthy of complexity to make it more efficient.     If we find that that in practice the 'Id' property is hot we can fix this.  

I have put a comment to that effect in the code.